### PR TITLE
Ana pwm

### PR DESCRIPTION
**Pull request by jgrizou**
Three examples for tutorials:
- breathing led using pwm
- brightness control via potentiometer
- servo control via potentiometer

### CODE REVIEW

File: examples/led_breathe.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// this will be changed in the furture for a more consistent API
+
+fn main() {
+    // initialize rcc
+    // it set some register in the microcontroller regarding frequency of timers
+    rcc::init();
+
+    // declare `led` as an output PWM
+    let led = pwm::Pwm::init(PIN_LED);
+    // Set the frequency of the PWM to 500 hz
```
Comment (matthieu-lapeyre) [inline examples/led_breathe.rs]: Why 500 ?

File: examples/led_potmeter.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+// define pin for the LED
+// PC7 is a led pin and we can set it up as PWM output
+const PIN_LED: pwm::Pin = pwm::Pin::PC7;
+
+// define pin for the analog read
+// we will use the pin PA0 on the board
+const PIN_ANALOG: adc::Pin = adc::Pin::PA0;
+// the adc are 12 bits so values range from 0 to 4095 (4096 values)
+const MAX_U12: f32 = 4095.0;
```
Comment (matthieu-lapeyre) [inline examples/led_potmeter.rs]: Should not it rather be an integer ?

File: examples/led_breathe.rs (hunk 1 above)
Comment (jgrizou) [reply examples/led_breathe.rs]: no reason, we just have to choose

File: examples/led_potmeter.rs (hunk 2 above)
Comment (jgrizou) [reply examples/led_potmeter.rs]: we can, and then transform it into a f32 when we use it

Comment (jgrizou) [reply examples/led_potmeter.rs]: fixed

File: examples/led_breathe.rs (hunk 1 above)
Comment (jgrizou) [reply examples/led_breathe.rs]: added line of explanation

**Comment by jgrizou** [comment]
updated the mock hal for compilation on linux target

**Pull request merged**